### PR TITLE
chore: OpenAPI spec changes & other support for running localenv in docker windows

### LINKED ISSUE #1562: Be able to run Rafiki in Windows Docker
Currently, there are a few issues that prevent running rafiki in docker windows. 

- [ ] Symlinks to the OpenAPI yaml specs are not properly referenced when the repo is cloned and used in windows (will be unblocked with https://github.com/interledger/open-payments/issues/439)
- [ ] `pnpm localenv:compose` commands do not work because of the `BEFORE_MERGED` usage
- [ ] Localenv in Tigerbeetle process is killed when starting up

### PR DESCRIPTION
**Pull request by mkurapov**
## Changes proposed in this pull request

- Update how we handle OpenAPI specs in the repo
- Remove the `BEFORE_MERGED` variable from our `localenv:compose` commands
- Add doc section to local playground overview with known issues

## Context

This PR makes localenv work inside docker for windows (whether that's through hyper-v or WSL). 

### Symlinked Schemas

The main issue that prevented this working previously were the symlinked OpenAPI schemas (in `packages/*/src/openapi`) that docker for windows couldn't copy over as a proper file when creating the docker images.
This led me down the road to a bigger refactor of how we were handling OpenAPI specs in the repo in general. 

Usually, we used the `fetch-schemas` script to pull the latest OpenAPI schemas into the root `openapi` folder, where we would from there, symlink to them from `packages/(backend | auth | token-introspection)/src/openapi`. We needed to do that for: 
1. Creating OpenAPI validators with (eg):
```ts
await createOpenAPI(
   path.resolve(__dirname, './openapi/resource-server.yaml')
)
```
2. Being able to have references to the schemas, so we could reference components in `auth-server.yaml`, for example, how we do in `token-introspection.yaml`:
https://github.com/interledger/rafiki/blob/96061ff72d20221e87cb21799ea6d68e95e261bd/packages/token-introspection/src/openapi/token-introspection.yaml#L70-L71

The first "problem" was solved by being able to directly import the OpenAPI objects from the updated `open-payments` library: 
- https://github.com/interledger/open-payments/pull/440

allowing us to do validation as such: https://github.com/interledger/rafiki/blob/aa556047edb5e6f6c7d267b2e9f1c56d2c86cfdd/packages/backend/src/index.ts#L157-L165

*This is handy because it allows us to remove the `fetch-schemas` script, and it means that the spec in the open payments client we use in the repo will always be consistent with the spec that we validate against when checking the Open Payment API requests and responses.*

However, this didn't fully solve the issue 2: how can we still have references to the yaml files such that we could extend them in our other OpenAPI files (namely, `token-introspection.yaml` and `id-provided.yaml`, which both reference the `auth-server.yaml` & shared `schemas.yaml`?

This was a bit tricky, and maybe there is a better solution, but this ends up working by using a `postinstall` command, that copies over the bundled yaml files in the `node_modules/@interledger/open-payments/...`folder into the respective `(auth | token-introspection)/src/openapi/specs` folder when doing `pnpm install`:

https://github.com/interledger/rafiki/blob/dceeb405991dc0f9dc7405b0f6f2a38873c1f02b/packages/token-introspection/package.json#L13-L15

Since this ends up being kind of a generated file, we should `.gitignore` them:
https://github.com/interledger/rafiki/blob/aa556047edb5e6f6c7d267b2e9f1c56d2c86cfdd/.gitignore#L60-L64

#### Question
Related to a suggestion from @BlairCurrey: should we have a sibling `openapi` package in the repo that basically holds all of our OpenAPI yaml files, such that we can export and use the the OpenAPI validators in `backend` and `auth` all from 
one spot?
This way, we can just do this `postinstall` command in one place to get the Open Payments specs. The downsides of this: the files that `backend` and `auth` use separately are _not_ in their corresponding repos and we would need to reference this package when generating the TS types for the `token-introspection` package.

### Other issues

Other issues that prevented running localenv in windows docker were:
- the `pnpm localenv:compose` commands do not work because of the `BEFORE_MERGED` variable usage
    - this was fixed just by writing out the commands in-line
- the tigerbeetle container was killed upon startup:
    - this was actually a documented [issue](https://docs.tigerbeetle.com/quick-start/with-docker-compose/#exited-with-code-137) , and its fixed just by increasing the available RAM in the VM. I added documentation for this.

Fixes #1562 

## Checklist

- [x] Related issues linked using `fixes #number`
- [x] Tests added/updated
- [x] Documentation added
- [ ] Make sure that all checks pass
- [ ] Bruno collection updated

### CODE REVIEW

File: localenv/mock-account-servicing-entity/Dockerfile [hunk 1]
```diff
@@ -7,10 +7,6 @@ ENV PATH="$PNPM_HOME:$PATH"
 
 RUN corepack enable
 RUN corepack prepare pnpm@8.7.4 --activate
-RUN apk add --no-cache \
-    python3 \
-    make \
-    g++
```
Comment (mkurapov) [inline localenv/mock-account-servicing-entity/Dockerfile:13]: was breaking on windows, and looks like not needed in general

File: packages/token-introspection/src/openapi/specs/token-introspection.yaml [hunk 2]
```diff
@@ -0,0 +1,116 @@
+openapi: 3.1.0
```
Comment (mkurapov) [inline packages/token-introspection/src/openapi/specs/token-introspection.yaml:1]: no changes, just moved under `openapi/specs` folder

File: localenv/mock-account-servicing-entity/Dockerfile (hunk 1 above)
Comment (BlairCurrey) [reply localenv/mock-account-servicing-entity/Dockerfile:13]: @raducristianpopa specified in the original pr that this is for `node-gyp`. https://github.com/interledger/rafiki/pull/1795#discussion_r1307418635 Could `node-gyp` be included with pnpm? It's liable to be used by some of our packages and its hard to say what. I did remove from backend and start localenv OK FWIW. This SO post suggests its part of npm at least https://stackoverflow.com/questions/67584602/does-node-gyp-come-with-npm

**Comment by BlairCurrey** [comment]
noticed when trying to setup on windows that `pnpm -r build` fails (see it on mac too). Looks like the `spec` dir gets created by `token-introspection` then fails when we try to do it again:

```console
blaircurrey@blairs-MacBook-Pro-2 rafiki % pnpm -r build
Scope: 6 of 7 workspace projects
localenv/mock-account-servicing-entity build$ remix build
│ [info] building... (NODE_ENV=production)
│ [info] built (340ms)
└─ Done in 2.3s
packages/token-introspection build$ pnpm clean && tsc --build tsconfig.json && pnpm copy-files
│ > token-introspection@ clean /Users/blaircurrey/code/interledger/rafiki/packages/token-introspection
│ > rm -fr dist/
│ > token-introspection@ copy-files /Users/blaircurrey/code/interledger/rafiki/packages/token-introspection
│ > mkdir ./dist/openapi/specs && cp -r ./src/openapi/specs/*.yaml ./dist/openapi/specs/
└─ Done in 1.1s
packages/frontend build$ remix build
│ [info] building... (NODE_ENV=production)
│ [info] built (770ms)
└─ Done in 2.3s
packages/auth build$ pnpm build:deps && pnpm clean && tsc --build tsconfig.json && pnpm copy-files
│ > auth@ build:deps /Users/blaircurrey/code/interledger/rafiki/packages/auth
│ > pnpm --filter token-introspection build
│ > token-introspection@ build /Users/blaircurrey/code/interledger/rafiki/packages/token-introspection
│ > pnpm clean && tsc --build tsconfig.json && pnpm copy-files
│ > token-introspection@ clean /Users/blaircurrey/code/interledger/rafiki/packages/token-introspection
│ > rm -fr dist/
│ > token-introspection@ copy-files /Users/blaircurrey/code/interledger/rafiki/packages/token-introspection
│ > mkdir ./dist/openapi/specs && cp -r ./src/openapi/specs/*.yaml ./dist/openapi/specs/
│ mkdir: ./dist/openapi/specs: File exists
│  ELIFECYCLE  Command failed with exit code 1.
│ /Users/blaircurrey/code/interledger/rafiki/packages/token-introspection:
│  ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  token-introspection@ build: `pnpm clean && tsc --build tsconfig.json && pnpm copy-files`
│ Exit status 1
│  ELIFECYCLE  Command failed with exit code 1.
└─ Failed in 1.5s at /Users/blaircurrey/code/interledger/rafiki/packages/auth
packages/backend build$ pnpm build:deps && pnpm clean && tsc --build tsconfig.json && pnpm copy-files
│ > backend@ build:deps /Users/blaircurrey/code/interledger/rafiki/packages/backend
│ > pnpm --filter token-introspection build
│ > token-introspection@ build /Users/blaircurrey/code/interledger/rafiki/packages/token-introspection
│ > pnpm clean && tsc --build tsconfig.json && pnpm copy-files
│ > token-introspection@ clean /Users/blaircurrey/code/interledger/rafiki/packages/token-introspection
│ > rm -fr dist/
│ > token-introspection@ copy-files /Users/blaircurrey/code/interledger/rafiki/packages/token-introspection
│ > mkdir ./dist/openapi/specs && cp -r ./src/openapi/specs/*.yaml ./dist/openapi/specs/
└─ Running...
/Users/blaircurrey/code/interledger/rafiki/packages/auth:
 ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  auth@ build: `pnpm build:deps && pnpm clean && tsc --build tsconfig.json && pnpm copy-files`
Exit status 1
```

**Comment by mkurapov** [comment]
@BlairCurrey got it, thanks! Updated the `mkdir` command

File: localenv/mock-account-servicing-entity/Dockerfile (hunk 1 above)
Comment (raducristianpopa) [reply localenv/mock-account-servicing-entity/Dockerfile:13]: There was a dependency chain within `testcontainers` that needed these packages (Pything and the C++ compiler) and I was following [their requirements](https://github.com/nodejs/node-gyp?tab=readme-ov-file#on-unix). I recall discussing with Max about this some time ago and it was working for me without installing them at that time.

**Review by sabineschaller — APPROVED** [review]
Is there something in the docs that talks about when to run the postinstall?

**Comment by mkurapov** [comment]
> Is there something in the docs that talks about when to run the postinstall?

`postinstall` runs automatically after every `pnpm i`: https://docs.npmjs.com/cli/v10/using-npm/scripts#life-cycle-operation-order

**Comment by mkurapov** [comment]
@sabineschaller 

> Is there something in the docs that talks about when to run the postinstall?

`postinstall` runs automatically after every `pnpm i`: https://docs.npmjs.com/cli/v10/using-npm/scripts#life-cycle-operation-order

**Comment by sabineschaller** [comment]
Of course 🤦‍♀️

**Pull request merged**